dataprocessors/json/json.go

```diff
@@ -0,0 +1,137 @@
+package json
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/spiceai/data-components-contrib/dataprocessors/json/observation"
+	"github.com/spiceai/spiceai/pkg/observations"
+	"github.com/spiceai/spiceai/pkg/state"
+	"github.com/spiceai/spiceai/pkg/util"
+	"go.skia.org/infra/go/jsonschema"
+)
+
+const (
+	JsonProcessorName string = "json"
+)
+
+type JsonProcessor struct {
+	data      []byte
+	dataMutex sync.RWMutex
+	dataHash  []byte
+	format    JsonFormat
+}
+
+type JsonFormat interface {
+	GetSchema() *[]byte
+	GetObservations(data *[]byte) ([]observations.Observation, error)
+	GetState(data *[]byte, validFields *[]string) ([]*state.State, error)
+}
+
+type ValidationError struct {
+	message         string
+	validationError string
+}
+
+func (e *ValidationError) Error() string {
+	return e.message
+}
+
+func NewJsonProcessor() *JsonProcessor {
+	return &JsonProcessor{}
+}
+
+func (p *JsonProcessor) Init(params map[string]string) error {
+	// Default format if not specified in params
+	format := "default"
+
+	if paramFormat, ok := params["format"]; ok {
+		format = paramFormat
+	}
+
+	switch format {
+	case "default":
+		p.format = &observation.ObservationJsonFormat{}
+	}
+
+	if p.format == nil {
+		return fmt.Errorf("unable to find json format '%s'", format)
+	}
+
+	return nil
+}
+
+func (p *JsonProcessor) OnData(data []byte) ([]byte, error) {
+	if p.format == nil {
+		return nil, fmt.Errorf("json processor not initialized")
+	}
+
+	p.dataMutex.Lock()
```

Comment:
Seems like it would be better to validate the schema before taking the lock and computing the hash. No point processing it if the schema is not valid.